src/main/java/org/opensearch/neuralsearch/processor/NeuralQueryProcessor.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor;
+
+import java.util.Map;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.common.Nullable;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.ingest.ConfigurationUtils;
+import org.opensearch.neuralsearch.query.visitor.NeuralSearchQueryVisitor;
+import org.opensearch.search.pipeline.AbstractProcessor;
+import org.opensearch.search.pipeline.Processor;
+import org.opensearch.search.pipeline.SearchRequestProcessor;
+
+/**
+ * Neural Search Query Request Processor, It modifies the search request with neural query clause
+ * and adds model Id if not present in the search query.
+ */
+@Setter
+@Getter
+public class NeuralQueryProcessor extends AbstractProcessor implements SearchRequestProcessor {
```

Comment:
I would rename class to something closer to a processor type name, although I left the final call to you